### PR TITLE
eduvpn-client: init at 4.2.0

### PR DESCRIPTION
**Pull request by jwijenbergh**
## Description of changes

This PR adds the eduVPN for Linux packages.

eduVPN is a VPN for education & research: https://eduvpn.org

Note: I also maintain the eduVPN client so I added myself to maintainers here too

Also added @benneti as maintainers, see #278995

Builds upon: #210180 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22714,6 +22714,8 @@ with pkgs;
 
   libedit = callPackage ../development/libraries/libedit { };
 
+  libeduvpn-common = callPackage ../development/libraries/libeduvpn-common { };
```
Comment (benneti) [inline pkgs/top-level/all-packages.nix:22717]: did you try to also put this in pkgs by-name (I think it should also work there)

**Review by benneti — COMMENTED** [review]
Builds and runs and might be the better if someone will also try to package the mac os version in nix.
If the go module works also in by-names I think it might be better there, apart from this the PR looks good to go from my side.

File: pkgs/development/libraries/libeduvpn-common/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildGoModule
+, fetchurl
+, gcc
+, pkg-config
```
Comment (benneti) [inline pkgs/development/libraries/libeduvpn-common/default.nix:5]: gcc and pkg-config are not needed for me to build and run

File: pkgs/development/libraries/libeduvpn-common/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  src = fetchurl {
+    url = "https://github.com/eduvpn/eduvpn-common/releases/download/${version}/eduvpn-common-${version}.tar.xz";
+    hash = "sha256-CqpOgvGGD6pW03fvKUzgoeCz6YgnzuYK2u5Zbw+/Ks4=";
+  };
+
+  vendorHash = null;
+
+  buildInputs = [ gcc ];
+
+  nativeBuildInputs = [ pkg-config ];
```
Comment (benneti) [inline pkgs/development/libraries/libeduvpn-common/default.nix:21]: see above

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (benneti) [reply pkgs/top-level/all-packages.nix:22717]: I just tried and you can simply move this `pkgs/development/libraries/libeduvpn-common/default.nix -> pkgs/by-name/li/libeduvpn-common/package.nix` and remove the appropriate line in `pkgs/top-level/all-packages.nix`

Comment (jwijenbergh) [reply pkgs/top-level/all-packages.nix]: done! thanks

File: pkgs/development/libraries/libeduvpn-common/default.nix (hunk 2 above)
Comment (jwijenbergh) [reply pkgs/development/libraries/libeduvpn-common/default.nix:5]: oh, pkg-config was indeed obvious that it wasn't needed. gcc not so much, I guess go already includes that

File: pkgs/development/libraries/libeduvpn-common/default.nix (hunk 3 above)
Comment (jwijenbergh) [reply pkgs/development/libraries/libeduvpn-common/default.nix:21]: done!!

**Review by benneti — APPROVED** [review]
From my side this is now good to go!

**Review by benneti — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Comment by jwijenbergh** [comment]
Thanks! @thiagokokada @benneti

**Comment by jwijenbergh** [comment]
Thanks! @thiagokokada @benneti

**Pull request merged**